src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -471,6 +513,43 @@ private void CompileNamedType(NamedTypeSymbol containingType)
             }
 
             var members = containingType.GetMembers();
+
+            // We first compile the accessors that contain 'field' keyword to avoid extra bindings for 'field' keyword when compiling other members.
+            for (int memberOrdinal = 0; memberOrdinal < members.Length; memberOrdinal++)
+            {
+                var member = members[memberOrdinal];
+
+                //When a filter is supplied, limit the compilation of members passing the filter.
+                if (member is not SourcePropertyAccessorSymbol { ContainsFieldKeyword: true } accessor ||
+                    !PassesFilter(_filterOpt, member))
+                {
+                    continue;
+                }
+
+                Debug.Assert(member.Kind == SymbolKind.Method);
+                Binder.ProcessedFieldInitializers processedInitializers = default;
+                CompileMethod(accessor, memberOrdinal, ref processedInitializers, synthesizedSubmissionFields, compilationState);
+            }
+
+            // After compiling accessors containing 'field' keyword, we mark the backing field of the corresponding property as calculated.
+            foreach (var member in members)
+            {
+                if (member is SourcePropertySymbolBase property)
+                {
+                    // PROTOTYPE(semi-auto-props): This can be optimized by checking for field keyword syntactically first.
+                    // If we don't have field keyword, we can safely ignore the filter check.
+                    // This will require more tests.
+                    var getMethod = property.GetMethod;
+                    var setMethod = property.SetMethod;
+                    if ((getMethod is null || PassesFilter(_filterOpt, getMethod)) &&
```

Comment:
>if ((getMethod is null || PassesFilter(_filterOpt, getMethod)) && [](http://example.com/codeflow?start=20&length=65)

Would it be worth incorporating ```ContainsFieldKeyword``` check for accessors into this ```if``` condition too?

Reply:
This is captured by a prototype comment above. Let me know if you'd like it to be in this PR.

Reply:
>This is captured by a prototype comment above. Let me know if you'd like it to be in this PR.

That is good enough. I missed this comment.